contracts/contracts/link_token.cairo

```diff
@@ -0,0 +1,83 @@
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.math import assert_not_zero, assert_le
+from starkware.cairo.common.uint256 import Uint256
+from starkware.starknet.common.syscalls import get_caller_address
+from starkware.cairo.common.bool import FALSE, TRUE
+from openzeppelin.token.erc20.library import ERC20
+from openzeppelin.token.erc20.presets.ERC20Upgradeable import (
+    initializer,
+    upgrade,
+    name,
+    symbol,
+    totalSupply,
+    decimals,
+    balanceOf,
+    allowance,
+    transfer,
+    transferFrom,
+    approve,
+    increaseAllowance,
+    decreaseAllowance,
+)
+from starkware.starknet.std_contracts.ERC20.permitted import (
+    permitted_initializer,
+    permitted_minter_only,
+    permittedMinter,
+)
+from contracts.ERC677.library import ERC677
+from contracts.ERC677.interfaces.IERC677Receiver import IERC677Receiver
+
+const NAME = 'ChainLink Token'
+const SYMBOL = 'LINK'
+
+@constructor
+func constructor{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    initial_supply : Uint256, recipient : felt, proxy_admin : felt
+):
+    alloc_locals
+    initializer(NAME, SYMBOL, 18, initial_supply, recipient, proxy_admin)
+    permitted_initializer(recipient)
+    return ()
+end
+
+@external
```

Comment:
Add a comment before these two functions saying this implements Starkgate IMintableToken interface